app/controllers/application_controller.rb

```diff
@@ -1,10 +1,17 @@
 class ApplicationController < ActionController::Base
+  include Pundit::Authorization
+
   protect_from_forgery with: :null_session
 
   before_action :authenticate_user!
 
   before_action :configure_permitted_parameters, if: :devise_controller?
 
+  rescue_from Pundit::NotAuthorizedError, ActiveRecord::RecordNotFound do |exception|
+    flash[:alert] = "You don't have access"
+    redirect_to root_path
+  end
```

Comment:
Could you please move logic to method ans use `rescue_from` in this way
```suggestion
  rescue_from Pundit::NotAuthorizedError, ActiveRecord::RecordNotFound, with: :method_name
```